### DISCUSSION TITLE
Nuxt plugin, feature and documentation

### DISCUSSION THREAD

**Discussion by JBustin** [histoire-dev/histoire#571] [General]
Hello,

I'm trying to figure out what the nuxt plugin covers.
My project runs on nuxt3 (migration from nuxt2 in progress), and I'm starting with histoire.
```
import { defineConfig } from 'histoire'
import { HstVue } from '@histoire/plugin-vue'
import { HstNuxt } from '@histoire/plugin-nuxt'

export default defineConfig({
  plugins: [HstVue(), HstNuxt()],
})
```

I'm a little bit surprise by some "issues" I experiment:

- `process.client` is not defined
- nuxt runtime config is missing
- window, navigator, document, ... globals from browser throws error during build (undefined), some of them related to the vue vendors
- I have to manually import some plugins, in despite of the fact that the build seems to parse them and throws the errors previously mentioned
- I cannot directly import nuxt plugin with the dedicated syntax `defineNuxtPlugin`, I need to use the vue plugin

I think that context and informations about the nuxt plugin could be useful.
Did I miss something ?

**Comment by riddla** [comment]
Nuxt plugins that are meant to be run on the client seem to have problems with missing DOM APIs like `requestAnimationFrame`, although JSDOM is configured with `pretendToBeVisual`.

In our case we are importing gsap in the client side plugin and `Error: Uncaught [ReferenceError: requestAnimationFrame is not defined]` is thrown.
